node/node.go

```diff
@@ -1630,6 +1759,45 @@ func makeNodeInfo(
 	return nodeInfo, err
 }
 
+func makeSeedNodeInfo(
+	config *cfg.Config,
+	nodeKey p2p.NodeKey,
+	genDoc *types.GenesisDoc,
+	state sm.State,
+) (p2p.NodeInfo, error) {
+	nodeInfo := p2p.NodeInfo{
+		ProtocolVersion: p2p.NewProtocolVersion(
+			version.P2PProtocol, // global
+			state.Version.Consensus.Block,
+			state.Version.Consensus.App,
+		),
+		NodeID:   nodeKey.ID,
+		Network:  genDoc.ChainID,
+		Version:  version.TMCoreSemVer,
+		Channels: []byte{},
+		Moniker:  config.Moniker,
+		Other: p2p.NodeInfoOther{
+			TxIndex:    "off",
+			RPCAddress: config.RPC.ListenAddress,
+		},
+	}
+
+	if config.P2P.PexReactor {
+		nodeInfo.Channels = append(nodeInfo.Channels, pex.PexChannel)
+	}
+
+	lAddr := config.P2P.ExternalAddress
+
+	if lAddr == "" {
+		lAddr = config.P2P.ListenAddress
+	}
+
+	nodeInfo.ListenAddr = lAddr
+
+	err := nodeInfo.Validate()
+	return nodeInfo, err
```

Comment:
should we return a non-nil node info when err is non nil?

Reply:
The assumption is generally to read the error first, so I think it should be fine